cmd/cluster-agent/api/v1/languagedetection/util_test.go

```diff
@@ -454,13 +465,199 @@ func TestOwnersLanguagesMergeAndFlush(t *testing.T) {
 	assert.False(t, ownersLanguages.containersLanguages[mockSupportedOwnerA].dirty, "deploymentA dirty flag should be reset to false")
 }
 
+func TestCleanExpiredLanguages(t *testing.T) {
+	mockSupportedOwnerA := langUtil.NewNamespacedOwnerReference("api-version", langUtil.KindDeployment, "deploymentA", "ns")
+	mockSupportedOwnerB := langUtil.NewNamespacedOwnerReference("api-version", langUtil.KindDeployment, "deploymentB", "ns")
+
+	mockStore := fxutil.Test[workloadmeta.Mock](t, fx.Options(
+		core.MockBundle(),
+		fx.Supply(workloadmeta.NewParams()),
+		workloadmeta.MockModuleV2(),
+	))
+
+	mockStore.Push(workloadmeta.SourceLanguageDetectionServer, workloadmeta.Event{
+		Type: workloadmeta.EventTypeSet,
+		Entity: &workloadmeta.KubernetesDeployment{
+			EntityID: workloadmeta.EntityID{
+				Kind: workloadmeta.KindKubernetesDeployment,
+				ID:   "ns/deploymentA",
+			},
+			DetectedLanguages: langUtil.ContainersLanguages{
+				*langUtil.NewContainer("some-container"): {
+					"python": {},
+					"java":   {},
+				},
+			},
+		},
+	},
+		workloadmeta.Event{
+			Type: workloadmeta.EventTypeSet,
+			Entity: &workloadmeta.KubernetesDeployment{
+				EntityID: workloadmeta.EntityID{
+					Kind: workloadmeta.KindKubernetesDeployment,
+					ID:   "ns/deploymentB",
+				},
+				DetectedLanguages: langUtil.ContainersLanguages{
+					*langUtil.NewContainer("some-container"): {
+						"python": {},
+						"java":   {},
+					},
+				},
+			},
+		},
+	)
+
+	ownersLanguages := OwnersLanguages{
+		containersLanguages: map[langUtil.NamespacedOwnerReference]*containersLanguageWithDirtyFlag{
+			mockSupportedOwnerA: {
+				languages: langUtil.TimedContainersLanguages{
+					*langUtil.NewContainer("some-container"): {
+						"python": time.Now().Add(-5 * time.Minute),
```

Comment:
might be useful to define this one in a var like `expiredTime` or similar and reuse it in all the tests.
same for `unexpiredTime`. I think it might help readability.